pipeline.py

```diff
@@ -72,31 +94,89 @@ class TextProcessingPipeline:
 
     def run(self):
         """
         Runs pipeline process scenario
         """
-        pass
+        articles = self.corpus_manager.get_articles().values()
+        cleaned_tokens = []
+        single_tagged = []
+        multiple_tagged = []
+        for article in articles:
+            raw_text = article.get_raw_text()
+            tokens = self._process(raw_text)
+            for token in tokens:
+                cleaned_tokens.append(token.get_cleaned())
+                single_tagged.append(token.get_single_tagged())
+                multiple_tagged.append(token.get_multiple_tagged())
+            article.save_as(' '.join(cleaned_tokens), 'cleaned')
+            article.save_as(' '.join(single_tagged), 'single_tagged')
+            article.save_as(' '.join(multiple_tagged), 'multiple_tagged')
 
     def _process(self, raw_text: str):
         """
         Processes each token and creates MorphToken class instance
         """
-        pass
+        cleaned_text = ' '.join(re.findall(r'[а-яёА-ЯЁ]+', raw_text))
+        analyzed_cleaned_text = Mystem().analyze(cleaned_text)
+        tokens = []
+        morph = pymorphy2.MorphAnalyzer()
+        for token in analyzed_cleaned_text:
+            if ('analysis' not in token) \
+                    or (not token['analysis']) \
+                    or ('lex' not in token['analysis'][0]
+                        or 'gr' not in token['analysis'][0]):
+                continue
+            morphological_token = MorphologicalToken(token['text'])
+            morphological_token.normalized_form = token['analysis'][0]['lex']
+            morphological_token.tags_mystem = token['analysis'][0]['gr']
+            morphological_token.tags_pymorphy = morph.parse(token['text'])[0].tag
+            tokens.append(morphological_token)
+        return tokens
 
 
 def validate_dataset(path_to_validate):
     """
     Validates folder with assets
     """
-    pass
+    dataset_path = Path(path_to_validate)
+    if not dataset_path.exists():
+        raise FileNotFoundError
+    if not dataset_path.is_dir():
+        raise NotADirectoryError
+    indices = []
+    digit_pattern = re.compile(r'\d+')
+    for files in dataset_path.glob('*'):
+        digit = digit_pattern.match(files.stem)
+        if not digit:
+            raise InconsistentDatasetError
+        index = int(digit.group())
+        if index not in indices:
+            indices.append(index)
+    if not indices:
+        raise EmptyDirectoryError
+    previous_index = 0
+    for index in indices:
+        if index - previous_index != 1 or indices[0] != 1:
+            raise InconsistentDatasetError
+        previous_index = index
+        raw_path = dataset_path / f'{index}_raw.txt'
+        meta_path = dataset_path / f'{index}_meta.json'
+        if not raw_path.exists() or not meta_path.exists():
+            raise InconsistentDatasetError
+        with open(raw_path, 'r', encoding='utf-8') as file:
```

Comment:
where did you find a requirement to open the file for emptiness? If this is not stated in docs, remove. Otherwise, please point to that place